src/strings-in-block/class-html.php

```diff
@@ -174,4 +174,34 @@ private function get_dom_handler( \WP_Block_Parser_Block $block ) {
 
 		return new StandardBlock();
 	}
+
+	/**
+	 * @param                        $text
+	 * @param                        $lang
+	 * @param \WP_Block_Parser_Block $block
+	 * @param array                  $string_translations
+	 * @param                        $element
+	 * @param                        $dom_handle
+	 *
+	 * @return \WP_Block_Parser_Block
+	 */
+	private function updateTranslationInBlock( $text, $lang, \WP_Block_Parser_Block $block, array $string_translations, $element, $dom_handle ) {
+		$translation                = null;
+		$translationFromPageBuilder = apply_filters( 'wpml_pb_update_translations_in_content', $text, $lang );
+		if ( $translationFromPageBuilder === $text ) {
+			$string_id = $this->get_string_id( $block->blockName, $text );
+			if ( Obj::path( [ $string_id, $lang, 'status'], $string_translations ) == ICL_TM_COMPLETE ) {
+				$translation = $string_translations[ $string_id ][ $lang ]['value'];
+			}
+		} else {
+			$translation = $translationFromPageBuilder;
+		}
```

Comment:
@BruceOnTheGo since both branches of the conditional set a value for $translation, it might be a good idea to extra the code into a `getTranslationFromPageBuilder` private method.

Something like that:

```php
private function updateTranslationInBlock( $text, $lang, \WP_Block_Parser_Block $block, array $string_translations, $element, $dom_handle ) {
	$translation = $this->getTranslationFromPageBuilder($text, $lang,$block,$string_translations);
	if ( $translation ) {
		$block = self::update_string_in_innerContent( $block, $element, $translation );
		$dom_handle->setElementValue( $element, $translation );
	}

	return $block;
}

private function getTranslationFromPageBuilder(  $text, $lang, \WP_Block_Parser_Block $block, array $string_translations ) {
	$translation                = null;
	$translationFromPageBuilder = apply_filters( 'wpml_pb_update_translations_in_content', $text, $lang );
	if ( $translationFromPageBuilder === $text ) {
		$string_id = $this->get_string_id( $block->blockName, $text );
		if ( Obj::path( [ $string_id, $lang, 'status' ], $string_translations ) == ICL_TM_COMPLETE ) {
			return $string_translations[ $string_id ][ $lang ]['value'];
		}
	}

	return $translationFromPageBuilder;
}
```

Reply:
@andreasciamanna I applied your suggestion with a small fix.